src/schema.ts

```diff
@@ -234,45 +234,34 @@ function timeUnitSummary(unit: TimeUnit, summary: Summary, type: Type) {
       });
   }
 
-  return binStats(bin, summary, type);
+  return binStats(bin, summary);
 }
 
 /**
  * @return a new summary based on a new binning scheme and old summary statistics
  */
-function binStats(bin, summary: Summary, type: Type): Summary {
+function binStats(bin, summary: Summary): Summary {
   // have the timeUnit bin scheme, need to determine new stats
   // start with summary, pre-binning
   const result = extend({}, summary);
 
-  const newUnique = binUnique(bin, summary.unique);
-  const values = uniqueToValues(newUnique);
-
-  result.unique = newUnique;
-  // 'count', 'valid', and 'missing' don't change
-  result.distinct = (type === Type.ORDINAL) ? keys(result.unique).length : (bin.stop - bin.start) / bin.step;
+  result.unique = binUnique(bin, summary.unique);
+  result.distinct = (bin.stop - bin.start) / bin.step;
   result.min = bin.start;
   result.max = bin.stop;
-  result.mean = dl.mean(values);
-  result.stdev = dl.stdev(values);
-  result.median = dl.median(values);
-  const quartiles = dl.quartile(values);
-  result.q1 = quartiles[0];
-  result.q2 = quartiles[2];
-  result.modeskew = dl.modeskew(values);
 
   return result;
 }
 
 /**
  * @return a new unique object based off of the old unique count and a binning scheme
  */
-function binUnique(bin, oldUnique) {
+function binUnique(bin: Bin, oldUnique) {
   const newUnique = {};
   for (var value in oldUnique) {
     // date types are tricky, need to convert into correct unit
-    let bucket = bin.unit.date ? bin.value(bin.unit.unit(new Date(value))) : bin.value(value);
-    if (!newUnique[bin.value(value)]) {
+    let bucket: any = bin.unit.date ? bin.value(bin.unit.unit(new Date(value))) : bin.value(Number(value));
```

Comment:
isn't it always number?  (Or I guess at most it can be a date?  If so, use union type rather than `any`).